core/config/routes.rb

```diff
@@ -2,10 +2,11 @@
 # see: https://github.com/rails/rails/issues/34872
 Rails.application.routes.draw do
   direct :cdn_image do |model, options|
-    options[:host] = Spree.cdn_host if Spree.cdn_host.present?
-    options[:host] ||= Rails.application.routes.default_url_options[:host]
+    opts = options.slice(:protocol, :host, :port)
```

Comment:
`options` have added default url options - `theme_id` and `theme_preview_id`. It was added to the generated URL. We need only `protocol`, `host` and `port`.